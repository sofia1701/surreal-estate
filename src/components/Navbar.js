import React from "react";
import { Link } from "react-router-dom";
import FacebookLogin from "react-facebook-login";
import PropTypes from "prop-types";
import logo from "../styles/nav-logo.png";
import "../styles/navbar.css";

export default function Navbar({ onLogin, userID, picture, onLogout, name }) {
  return (
    <>
      <div className="logo-container">
        <Link to="/" className="logo">
          <img
            data-testid="logo-id"
            className="logo-img"
            alt="logo"
            src={logo}
          />
        </Link>
      </div>
      <div className="navbar">
        <ul className="menu">
          <li>
            <Link
              to="/properties"
              className="item"
              data-testid="view-properties-id"
            >
              View Properties
            </Link>
          </li>
          <li>
            <Link
              to="/add-property"
              className="item"
              data-testid="add-property-id"
            >
              Add a Property
            </Link>
          </li>
        </ul>
      </div>
      <div className="facebook-container">
        {userID ? (
          <button
            onClick={onLogout}
            className="my-facebook-button-class"
            type="submit"
          >
            Welcome,
            <br />
            {name}
            <br />
            Sign Out
            <img src={picture} alt="profile" />
          </button>
        ) : (
          <FacebookLogin
            appId="651558712097637"
            fields="name,email,picture"
            callback={onLogin}
            cssClass="my-facebook-button-class"
          />
        )}
      </div>
    </>
  );
}

Navbar.propTypes = {
  onLogin: PropTypes.func.isRequired,
  userID: PropTypes.string.isRequired,
  onLogout: PropTypes.func.isRequired,
  picture: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
};
