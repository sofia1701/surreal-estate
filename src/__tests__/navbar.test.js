import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Navbar from "../components/Navbar";

describe("Navbar", () => {
  it("renders correctly", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the correct specs", () => {
    const { getByTestId } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>
    );
    const logo = getByTestId("logo-id");
    expect(logo).toBeInTheDocument();

    const viewProperties = getByTestId("view-properties-id");
    expect(viewProperties).toHaveTextContent("View Properties");

    const addProperty = getByTestId("add-property-id");
    expect(addProperty).toHaveTextContent("Add a Property");
  });
});
