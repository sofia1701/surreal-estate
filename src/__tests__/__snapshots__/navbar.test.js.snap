// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar renders correctly 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <a
      href="/"
    >
      <img
        alt="logo"
        class="logo"
        data-testid="logo"
        src="nav-logo.png"
      />
    </a>
    <ul
      class="navbar-links"
    >
      <li>
        <a
          class="navbar-links-item"
          href="/"
        >
          View Properties
        </a>
      </li>
      <li>
        <a
          class="navbar-links-item"
          href="/add-property"
        >
          Add a Property
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
